replica/compaction_group.hh

```diff
@@ -217,6 +217,9 @@ public:
     //  2) Compacts all sstables which overlap with the split point
     // Returns a future which resolves when this process is complete.
     future<> split(compaction_group_list&, sstables::compaction_type_options::split opt);
+
+    // Make an sstable set spanning all sstables in the storage_group
+    lw_shared_ptr<sstables::sstable_set> make_sstable_set() const;
```

Comment:
Maybe return a const sstable_set? can be done later.

We should avoid making empty sstable sets and adding to them later, and try to have only sealed sets, where possible.

Reply:
At least the compound_sstable_set and tablet_sstable_sets that are returned on this path do not implement insert and erase so they are effectively immutable.